qemu/tests/migration.py

```diff
@@ -198,7 +202,7 @@ def check_dma():
             # run some functions before migrate start.
             pre_migrate = get_functions(params.get("pre_migrate"), locals())
```

Comment:
move get_functions out of run introduced a regression issue:
pre_migrate and post_migrate is none if other tests run migration as a subtest via run_virt_sub_test().
pre_migrate and post_migrate can not be got via locals().

@ldoktor 
Would you please resolve the regression issue?
Thanks.

Reply:
Hello @lijinlijin, I'm not sure what regression you have in mind, do you have a reproducer? The `locals` was used before, I haven't changed that part. I only allow to re-use the existing function from other tests.

Reply:
When I run balloon_in_use.with_live_migration.during_bg_test case, I found the pre_migrate was not executed at all.
The reason is "f = locals_dict.get(f)" is none in  get_functions().
Before your moving, pre_migrate can be got via locals(), after the moving it cannot.
I tried get pre_migrate from globals(), it works.
But I'm not sure whether it will introduce other issues like function name conflict in globals(), so I'd like to see if you have better ideas .

Reply:
I see and it makes sense. I wanted to avoid the use of globals to minimize the changes and  avoid potential problems, but it shouldn't really break anything and it fixes the issue when running on imported function. I'm sorry for introducing this regression, sending fix now...

Reply:
https://github.com/autotest/tp-qemu/pull/1778